app/javascript/App/bundles/CoinShow/HistoricalPriceDataTable.tsx

```diff
@@ -0,0 +1,236 @@
+import * as moment from 'moment'
+import { Moment } from 'moment'
+import DateRangeSelect from './DateRangeSelect'
+
+enum STATUSES {
+  INITIALIZING = 'INITIALIZING',
+  LOADING = 'LOADING',
+  READY = 'READY',
+}
+
+interface RawPriceData {
+  coin_key: string
+  to_currency: string
+  time: string
+  open: number
+  high: number
+  low: number
+  close: number
+  volume_from: number // volume in present currency
+  volume_to: number // volume in usd (or to_currency?)
+}
+
+interface PriceData {
+  time: string
+  open: string
+  high: string
+  low: string
+  close: string
+  volume: string
+  formattedTime: string
+  datetime: Moment
+  marketCap: string
+}
+
+interface Props {
+  classes: any
+  initialData?: RawPriceData[]
+  availableSupply: number
+  symbol: string
+}
+
+interface State {
+  data: PriceData[]
+  status: string
+  currency?: string
+  start?: Moment
+  end?: Moment
+}
+
+const styles = (theme) =>
+  createStyles({
+    table: {},
+    toolbar: {},
+    grow: {
+      flexGrow: 1,
+    },
+    picker: {},
+  })
+
+class HistoricalPriceDataTable extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    const initialSortedData = props.initialData
+      ? props.initialData
+          .map<PriceData>(this.parseData.bind(this))
+          .sort(this.sortDataFunc)
+      : undefined
+
+    // Set initial status
+    const statusIsReady = !_.isUndefined(props.initialData)
+    const initialStatus = statusIsReady ? STATUSES.READY : undefined
+    const currency = props.initialData
+      ? _.get(props, ['initialData', '0', 'to_currency'])
+      : undefined
+
+    this.state = {
+      status: initialStatus || STATUSES.INITIALIZING,
+      data: initialSortedData || [],
+      currency,
+    }
+  }
+
+  public sortDataFunc(x: PriceData, y: PriceData) {
+    return y.datetime.diff(x.datetime)
+  }
+
+  public formatPrice(price: number, decimal: number = 6): string {
+    return price.toLocaleString('en-US', {
+      maximumFractionDigits: decimal,
+    })
+  }
+
+  public parseData(d: RawPriceData): PriceData {
+    const datetime = moment.utc(d.time)
+    const formattedTime = datetime.format('MMM DD, YYYY')
+    const marketCap = this.formatPrice(
+      Math.round(this.props.availableSupply * d.close),
+      0,
+    )
+    const open = this.formatPrice(d.open)
+    const high = this.formatPrice(d.high)
+    const low = this.formatPrice(d.low)
+    const close = this.formatPrice(d.close)
+    const volume = this.formatPrice(d.volume_from)
+
+    return {
+      open,
+      high,
+      low,
+      close,
+      volume,
+      time: d.time,
+      datetime,
+      formattedTime,
+      marketCap,
+    }
+  }
+
+  public onDateChangeHandler = (startDate: string, endDate: string) => {
+    this.setState({
+      start: startDate ? moment(startDate) : null,
+      end: endDate ? moment(endDate) : null,
+    })
+  }
+
+  public componentDidMount() {
```

Comment:
@anthotsang Is this method going to be used in the future?

Reply:
I was setting the component up assuming that it might not have the data up front. But I haven't actually bothered filling out all the async parts yet since it's not needed at this time. I can remove it for now if you'd like.

Reply:
I'm okay to leave it for now as long as we remember, thanks for the heads up!